src/Methods.cs

```diff
@@ -1000,6 +1000,41 @@ public List<LibraryInfo> HandleDeprecateMSFTLicense(List<LibraryInfo> libraries)
             }
         }
 
+        public void ConvertHtmlFileToText(string htmlFile)
+        {
+            var htmlDocument = new HtmlAgilityPack.HtmlDocument();
+            
+            htmlDocument.Load(htmlFile);
+            
+            string htmlStrippedText =
+                System.Web.HttpUtility.HtmlDecode(
+                    htmlDocument.DocumentNode.InnerText
+                );
+            
+            string textFile = htmlFile.Replace(".html", ".txt");
+            
+            WriteOutput(
+                line: $"Converting {htmlFile} to {textFile}...",
+                logLevel: LogLevel.Verbose
+            );    
+            File.WriteAllText(textFile, htmlStrippedText);
+
+            WriteOutput(
+                line: $"Deleting {htmlFile}...",
+                logLevel: LogLevel.Verbose
+            );
+            File.Delete(htmlFile);
+        }
+
+        public void ConvertHtmlFilesToText(string htmlFolder)
+        {
+            var htmlFiles = Directory.GetFiles(htmlFolder, "*.html");
```

Comment:
Hi @unsolaci,

Instead of saving all HTML files and then converting them into text files, would you mind intercepting the stream at :990 and saving converted txt directly?

I think this would eliminate unnecessary file management.

Thanks!

Reply:
Hey @Lexy2,

Your suggestion does sound like a good idea, but I'm unsure about the
implementation.

[HtmlAgilityPack](https://github.com/zzzprojects/html-agility-pack/) provides
methods that work with files and strings. I don't know on how to work with that
asynchronously in the context of `ExportLicenseTexts()`.

I could simply do something like this:
```csharp
public string ConvertHtmlToText(string html) {
    var htmlDocument = new HtmlAgilityPack.HtmlDocument();
    
    htmlDocument.LoadHtml(html);
    
    string htmlStrippedText =
        System.Web.HttpUtility.HtmlDecode(
            htmlDocument.DocumentNode.InnerText
        );
    
    return htmlStrippedText;
}
```

And then inside `ExportLicenseTexts()`:
```csharp
string licenseFileContents =
    await response.Content.ReadAsStringAsync();

string htmlStrippedLicenseFileContents =
    ConvertHtmlToText(licenseFileContents);

System.IO.File.WriteAllText(
    path: outpath, 
    contents: htmlStrippedLicenseFileContents
);
```
But I have a feeling that this might be defeating the purpose of originally
using async here.

On top of that, if `--convert-html-to-text` is supposed to be an option, then
I'd also have to check for the option from within `ExportLicenseTexts()`.

Reply:
Hey, I like the idea with `ReadAsStringAsync`!

If you're worried about asyncness further down the road, use `StreamWriter.WriteAsync`.

The question though is: how do you ensure that the license text is not stripped during conversion, like the ones from opensource.org?
Have you tested HtmlAgilityPack in these tough cases?
In this case, it should definitely be a separate experimental option as we can't guarantee that HtmlAgilityPack does the correct conversion.

Reply:
And best to have a test for this option.

Reply:
> The question though is: how do you ensure that the license text is not stripped during conversion, like the ones from opensource.org?
> Have you tested HtmlAgilityPack in these tough cases?

I've tried it with https://opensource.org/licenses/MIT and while the output
doesn't look particularly pretty (some non-license text from the page remains),
the license text itself is *not* stripped off.


> In this case, it should definitely be a separate experimental option as we can't guarantee that HtmlAgilityPack does the correct conversion.

I'm completely fine with this being marked as an experimental option. While not
very sophisticated, this implementation is about as much as I can contribute at
this time. Any improvements are of course welcome.


> On top of that, if --convert-html-to-text is supposed to be an option, then
> I'd also have to check for the option from within ExportLicenseTexts().

Do you have any suggestions on how I should go about checking whether the
`--convert-html-to-text` option has been passed from within 
`ExportLicenseTexts()`?